### PR TITLE
UIOR-1035 UIOR-1036 UIOR-1057 Display selected version in PO and POL view

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UIOR-1035
https://issues.folio.org/browse/UIOR-1036
https://issues.folio.org/browse/UIOR-1057
## Approach

- Add permissions and interfaces to fetch PO and POL audit events
- Clean up mock data
- Add hooks and utils for fetching PO and POL versions data
- Create components for display PO and POL version details

Demo (rancher env): https://folio-dev-thunderjet-diku.ci.folio.org/orders/view/c3fc5071-ec6e-4caa-b950-76b975388763/versions/79d9fb11-13c7-4969-965a-619d27ee847f?limit=50&offset=0&workflowStatus=Open&workflowStatus=Pending

<details>
  <summary>Preview</summary>

chrome_kG1VB9Yyz9

</details>

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by zburke — CHANGES_REQUESTED** [review]
Adding new required okapi interfaces requires bumping the major version of the application.

**Comment by usavkov-epam** [comment]
> Adding new required okapi interfaces requires bumping the major version of the application.

Thanks Zak! Updated `package.json`

**Review by uladislausamets — APPROVED** [review]

**Comment by NikitaSedyx** [comment]
hi @zburke @Dmitriy-Litvinenko could you please take a look?

File: src/common/VersionView/VersionCheckbox/VersionCheckbox.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      disabled
+      label={checkboxLabel}
+      vertical
+      {...props}
+    />
+  );
+};
+
+VersionCheckbox.propTypes = {
+  checked: PropTypes.bool,
```
Comment (Dmitriy-Litvinenko) [inline src/common/VersionView/VersionCheckbox/VersionCheckbox.js:25]: Could you please investigate can we add default value for not required attribute across this pull request?

File: src/components/POLine/Location/LocationView.js [hunk 2]
```diff
@@ -23,14 +29,20 @@ const Location = ({ location, locationsMap, holdings }) => {
   const locationValue = location.holdingId
     ? getHoldingLocationName(holding, locationsMap)
     : locationNameCode;
+  const locationName = location.holdingId
+    ? `${fieldName}.holdingId`
```
Comment (Dmitriy-Litvinenko) [inline src/components/POLine/Location/LocationView.js:33]: name not required (name: PropTypes.string,) here and we can get problem. Could you please investigate this problem and similar problem across this pull?

File: src/components/POLine/POLineVersionView/VendorVersionView/VendorVersionView.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+} from '@folio/stripes/components';
+import { VendorReferenceNumbersDetails } from '@folio/stripes-acq-components';
+
+import { VersionKeyValue } from '../../../../common/VersionView';
+
+export const VendorVersionView = ({ version }) => {
+  const vendorDetail = version?.vendorDetail;
+
+  return (
+    <>
```
Comment (Dmitriy-Litvinenko) [inline src/components/POLine/POLineVersionView/VendorVersionView/VendorVersionView.js:17]: Could you please investigate do we need this tag <> at all?

File: src/components/PurchaseOrder/POVersionView/OngoingInfoVersionView/OngoingInfoVersionView.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import {
+  VersionCheckbox,
+  VersionKeyValue,
+} from '../../../../common/VersionView';
+
+export const OngoingInfoVersionView = ({ version }) => {
+  const data = version?.ongoing;
+
+  return (
+    <>
```
Comment (Dmitriy-Litvinenko) [inline src/components/PurchaseOrder/POVersionView/OngoingInfoVersionView/OngoingInfoVersionView.js:19]: Could you please investigate do we need this tag <> at all?

File: src/common/VersionView/VersionCheckbox/VersionCheckbox.js (hunk 1 above)
Comment (usavkov-epam) [reply src/common/VersionView/VersionCheckbox/VersionCheckbox.js:30]: Add default values for `VersionCheckbox` and `VersionKeyValue` boolean props, the rest is ok as `undefined`

File: src/components/POLine/Location/LocationView.js (hunk 2 above)
Comment (usavkov-epam) [reply src/components/POLine/Location/LocationView.js]: Good catch! The absence of the 'name' prop will result in a confusing name being assigned. Thanks!

File: src/components/POLine/POLineVersionView/VendorVersionView/VendorVersionView.js (hunk 3 above)
Comment (usavkov-epam) [reply src/components/POLine/POLineVersionView/VendorVersionView/VendorVersionView.js]: react element removed as redundant

File: src/components/PurchaseOrder/POVersionView/OngoingInfoVersionView/OngoingInfoVersionView.js (hunk 4 above)
Comment (usavkov-epam) [reply src/components/PurchaseOrder/POVersionView/OngoingInfoVersionView/OngoingInfoVersionView.js]: react element removed as redundant

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Pull request merged**